test/NuGet.Core.Tests/NuGet.CommandLine.Xplat.Tests/Commands/Why/DependencyGraphFinderTests.cs

```diff
@@ -206,6 +206,32 @@ public void DifferentCaseUsedForTargetPackageId_MatchesAreCaseInsensitive_AllPat
             Assert.Contains(dependencyGraphs["net472"].First(dep => dep.Id == "DotnetNuGetWhyPackage").Children, dep => (dep.Id == "System.Text.Json") && (dep.Version == "8.0.0"));
         }
 
+        [Fact]
+        public void GetAllDependencyGraphsForTarget_WithWinX64Runtime_FindsRIDSpecificPackages()
```

Comment:
I think we're not testing enough scenarios here. 

You're introducing the search in graphs with rids and you're only adding a success case where the package exists in a rid. 

Some other scenarios that should be tested: 

1. Ridless graph has 1 version, ridfull graph has another version.
2. multi targetted project, at least 2x frameworks, at least 2x rids, combinations of package being found in 0, 1 or both graphs with rids. 
3. The package is not available at all in a multi-rid project. I think this catches the bug I pointed out above. 
4. Casing tests. If I change line 53 in DependencyGraphFinder, no tests likely fail.